src/neo4clj/cypher.clj

```diff
@@ -64,18 +66,37 @@
        (when props (properties props))
        "]->" to ))
 
-(defn lookup
-  "Takes a lookup representation and returns its cypher equvalent
+(defn lookup-where
+  [{:keys [ref-id id props]}]
+  (cond
+    id (str "ID(" ref-id ") = " id)
+    props (where ref-id props)
+    :else nil))
+
+(defn lookup-node
+  "Takes a node lookup representation and returns its cypher equvalent
 
   The return value is an vector with the first part being the actual entity and
   the second is the where clause for the lookup."
-  [{:keys [id ref-id props] :as lookup}]
+  [{:keys [ref-id id props] :as lookup}]
   (vector (str "(" ref-id (labels (:labels lookup))
                (when (map? props) (properties props))
                ")")
-          (when-let [where-parts (not-empty
-                                  (remove nil? [(when id
-                                                  (str "ID(" ref-id ") = " id))
-                                                (when (and props (not (map? props)))
-                                                  (str "(" (where ref-id props) ")"))]))]
-            (str/join " AND " where-parts))))
+          (when (or id (not (map? props)))
```

Comment:
I'm trying to find the reason for putting this conditional here. `lookup-where` has its own conditionals inside a `cond` and it returns `nil` when nothing matches (just like `when`). Why not always call `lookup-where` and sometimes it will return `nil`?

Reply:
The reason for this is, if there is no id and the properties are a map, then they are already included in the Nodes match part. So the where should only be added for id or complex AND / OR poperties.

But I can see, that I might add logic to remove the properties totally when an id is given to improve performance.